cli/jhipster.js

```diff
@@ -35,21 +35,31 @@ if (!semver.satisfies(currentNodeVersion, minimumNodeVersion)) {
     process.exit(1);
 }
 
-let preferGlobal = false;
-if (process.argv.length >= 3) {
-    if (process.argv[2] === 'upgrade') {
-        // Prefer global version for `jhipster upgrade` to get most recent code
-        preferGlobal = true;
-    }
-}
-requireCLI(preferGlobal);
+let preferLocal = true;
+
+program.command('upgrade')
```

Comment:
I was wrong: under debugger I was able to see that `program` is a global variable and in cl.js it is still holding the options set in jhipster.js. In fact, only the ref to the object is a const. I will rollback to previous code that did not use commander.